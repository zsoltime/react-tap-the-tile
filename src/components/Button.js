import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components/macro';

const StyledButton = styled.button.attrs(props => ({
  type: props.type ? props.type : 'button',
}))`
  background-color: ${props =>
    props.primary ? '#46b980' : '#363636'};
  border: 0;
  border-radius: 0.25rem;
  color: ${props => (props.primary ? 'inherit' : '#fff')};
  font-size: 1.5rem;
  max-width: 12rem;
  padding: 0.75rem 1rem;
  user-select: none;
  width: 100%;

  &:focus {
    outline: thin dotted;
  }
`;

const Button = ({ children, className, onClick, ...props }) => (
  <StyledButton className={className} onClick={onClick} {...props}>
    {children}
  </StyledButton>
);

Button.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
  onClick: PropTypes.func.isRequired,
};

export default Button;
