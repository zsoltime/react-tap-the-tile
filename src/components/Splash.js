import React from 'react';
import styled from 'styled-components/macro';

import Button from './Button';
import Logo from '../icons/Logo';
import { ModalText, Title } from './Common.style';

const BlackOnWhite = styled.strong`
  border: 1px solid #363636;
  padding: 2px 5px;
`;

const StyledLogo = styled(Logo)`
  margin-bottom: 1rem;
`;

const WhiteOnBlack = styled.strong`
  background-color: #363636;
  color: #fff;
  padding: 2px 5px;
`;

const Splash = ({ onStart }) => (
  <>
    <Title>Tap the Tile</Title>
    <StyledLogo />
    <ModalText>
      <p>
        Collect points by clicking on the{' '}
        <WhiteOnBlack>black</WhiteOnBlack> tiles before they fade out.
      </p>
      <p>
        Don't tap the <BlackOnWhite>white</BlackOnWhite> tiles.
      </p>
    </ModalText>
    <Button onClick={onStart} primary>
      New Game
    </Button>
  </>
);

export default Splash;
