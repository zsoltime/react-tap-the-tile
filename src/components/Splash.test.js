import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';

import Button from './Button';
import Splash from './Splash';
import { ModalText } from './Common.style';

describe('<Splash />', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(<Splash onStart={f => f} />);

    expect(wrapper.find(ModalText).render()).toMatchSnapshot();
  });

  it('calls onStart on click', () => {
    const onStart = jest.fn();
    const wrapper = shallow(<Splash onStart={onStart} />);
    wrapper.find(Button).simulate('click');

    expect(onStart).toBeCalledTimes(1);
  });
});
