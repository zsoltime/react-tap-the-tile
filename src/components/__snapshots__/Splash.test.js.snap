// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Splash /> matches snapshot 1`] = `
.c0 {
  font-size: 1.25rem;
  font-weight: 300;
  line-height: 1.25;
  margin-bottom: 1rem;
  text-align: center;
}

.c0 p {
  margin: 0 0 1rem;
}

.c2 {
  border: 1px solid #363636;
  padding: 2px 5px;
}

.c1 {
  background-color: #363636;
  color: #fff;
  padding: 2px 5px;
}

<div
  class="c0"
>
  <p>
    Collect points by clicking on the 
    <strong
      class="c1"
    >
      black
    </strong>
     tiles before they fade out.
  </p>
  <p>
    Don't tap the 
    <strong
      class="c2"
    >
      white
    </strong>
     tiles.
  </p>
</div>
`;
